zulip/tests/test_api.py

```diff
@@ -0,0 +1,95 @@
+import json
+import unittest
+import responses
+import zulip
+import urllib
+
+from typing import Any, Tuple, Dict
+from unittest import TestCase
+
+class TerminationException(Exception):
+    pass
+
+class TestAPI(TestCase):
+
+    @responses.activate
+    def test_add_reaction(self) -> None:
+        def request_callback(request: Any) -> Tuple[int, Dict[str, str], str]:
+            params = {}
+            for param in request.body.split("&"):
+                key, value = param.split("=")
+                params[key] = urllib.parse.unquote(value)
+            assert "emoji_name" in params or "emoji_code" in params
+            return (200, {}, json.dumps({'result': 'success', 'msg': ''}))
+
+        responses.add_callback(
+            method=responses.POST,
+            url="https://test.zulipapi.com/api/v1/messages/10/reactions",
+            callback=request_callback
+        )
+
+        client = zulip.Client(config_file="zulip/tests/test_zuliprc")
+        # request with emoji name
+        request = {
+            "message_id": 10,
+            "emoji_name": "octopus",
+        }
+        result = client.add_reaction(request)
+        self.assertEqual(result, {'result': 'success', 'msg': ''})
+        # request with emoji code
+        request = {
+            "message_id": 10,
+            "emoji_code": "1f419",
+        }
+        result = client.add_reaction(request)
+        self.assertEqual(result, {'result': 'success', 'msg': ''})
+
+    @responses.activate
+    def test_call_on_each_event(self) -> None:
```

Comment:
I think this test could benefit from a comment or docstring explaining what we're trying to test here.